Sources/MapboxCoreNavigation/EventDetails.swift

```diff
@@ -95,7 +95,13 @@ struct NavigationEventDetails: EventDetails {
     let simulation: Bool
     let startTimestamp: Date?
     let sdkIdentifier: String
-    let sdkVersion: String = String(describing: Bundle.mapboxCoreNavigation.object(forInfoDictionaryKey: "CFBundleShortVersionString")!)
+    var sdkVersion: String {
+        if let stringForShortVersion = Bundle.mapboxCoreNavigation.object(forInfoDictionaryKey: "CFBundleShortVersionString") {
+            return String(describing: stringForShortVersion)
+        } else {
+            return String(describing: UserDefaults.standard.object(forKey: "CFBundleShortVersionString"))
```

Comment:
If the Core Navigation bundle can’t be found and the user default isn’t set, wrapping the `nil` user default in a string would result in the string “`nil`”, which would be unexpected. The previous code would’ve crashed in that circumstance, so it would be reasonable to issue a `preconditionFailure(_:)` instead.

Reply:
Thanks for the suggestion, I'll add the check of this one.